src/nagini_translation/translators/common.py

```diff
@@ -257,15 +257,20 @@ def get_may_set_predicate(self, rec: Expr, field: PythonField, ctx: Context,
         pred_acc = self.viper.PredicateAccessPredicate(pred, full_perm, pos, info)
         return pred_acc
 
+    def check_var_defined(self, target: PythonVar, position: Position,
```

Comment:
Though it is interesting to check for var definition, mypy can do it. Is there any reason to do this check explicitly here?

Reply:
Yes, Mypy checks definedness optimistically, so Mypy's check is not sound.